### PR TITLE
feat(vec): introduce slice(), take() and drop() functions

### PR DESCRIPTION
**Pull request by jrmajor**

### CODE REVIEW

File: src/Psl/Dict/drop.php [hunk 1]
```diff
@@ -9,8 +9,8 @@
  *
  * Examples:
  *
- *      Dict\drop([1, 2, 3, 4, 5], 3)
- *      => Dict(4, 5)
+ *      Dict\drop(['a' => 1, 'b' => 2, 'c' => 3, 'd' => 4], 2)
+ *      => Dict('c' => 3, 'd' => 4)
```
Comment (jrmajor) [inline src/Psl/Dict/drop.php:13]: I've used the old examples from `Dict\drop()` and `Dict\slice()` in their `Vec` counterparts, I think they make more sense there.

File: src/Psl/Internal/Loader.php [hunk 2]
```diff
@@ -121,6 +121,7 @@ final class Loader
         'Psl\Vec\chunk',
         'Psl\Vec\chunk_with_keys',
         'Psl\Vec\concat',
+        'Psl\Vec\drop',
```
Comment (jrmajor) [inline src/Psl/Internal/Loader.php:124]: Not sure what's the correct order for these, they don't seem to be sorted alphabetically.

File: src/Psl/Vec/slice.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * @param iterable<T> $iterable Iterable to take the slice from
+ * @param int<0, max> $start Start offset
+ * @param null|int<0, max> $length Length (if not specified all remaining values from the array are used)
+ *
+ * @return list<T>
+ */
+function slice(iterable $iterable, int $start, ?int $length = null): array
+{
+    $result = [];
```
Comment (azjezz) [inline src/Psl/Vec/slice.php:28]: todo (for future PR if you would like): if `$iterable` is array, use `array_values(array_slice(...))` ( we would need to add a benchmark to see if there's a benefit first )

File: src/Psl/Internal/Loader.php (hunk 2 above)
Comment (azjezz) [reply src/Psl/Internal/Loader.php:124]: it's random :p

**Review by azjezz — APPROVED** [review]

File: src/Psl/Vec/take.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+declare(strict_types=1);
+
+namespace Psl\Vec;
+
+/**
+ * Take the first n elements from an iterable.
+ *
+ * Examples:
+ *
+ *      Vec\drop([1, 2, 3, 4], 2)
```
Comment (bendavies) [inline src/Psl/Vec/take.php:12]: wrong example

File: src/Psl/Dict/take.php [hunk 5]
```diff
@@ -7,6 +7,11 @@
 /**
  * Take the first n elements from an iterable.
  *
+ * Examples:
+ *
+ *      Dict\drop(['a' => 1, 'b' => 2, 'c' => 3, 'd' => 4], 2)
```
Comment (bendavies) [inline src/Psl/Dict/take.php:12]: wrong example

File: src/Psl/Vec/take.php (hunk 4 above)
Comment (jrmajor) [reply src/Psl/Vec/take.php:12]: Fixed in #354.

File: src/Psl/Dict/take.php (hunk 5 above)
Comment (jrmajor) [reply src/Psl/Dict/take.php:12]: Fixed in #354.

**Pull request merged**